vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/test/grid-connector-tree.test.ts

```diff
@@ -173,11 +175,29 @@ describe('grid connector - tree', () => {
 
     // Add a child item
     const childItem = { key: '1', name: 'foo bar', children: false };
-    setChildItems(grid.$connector, rootItem, [childItem]);    
+    setChildItems(grid.$connector, rootItem, [childItem]);
 
     // Add the expanded root item
     setRootItems(grid.$connector, [rootItem]);
 
     expect(getBodyCellText(grid, 1, 0)).to.equal('foo bar');
   });
+
+  it('should not request items for expanded parent outside the DOM', async () => {
```

Comment:
Tests that [hasCacheForParentKey](https://github.com/vaadin/flow-components/blob/182a875769714107f619d5ef4cfabb31a6c57f1e/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js#L126) works as expected